common/credential-storage/src/ephemeral_storage.rs

```diff
@@ -53,31 +53,41 @@ impl Storage for EphemeralStorage {
 
     async fn get_next_unspent_credential(
         &self,
-        gateway_id: &str,
     ) -> Result<Option<StoredIssuedCredential>, Self::StorageError> {
         // first try to get a free pass if available, otherwise fallback to bandwidth voucher
         let maybe_freepass = self
             .coconut_credential_manager
-            .get_next_unspect_freepass(gateway_id)
+            .get_next_unspent_freepass()
             .await;
         if maybe_freepass.is_some() {
             return Ok(maybe_freepass);
         }
 
         Ok(self
             .coconut_credential_manager
-            .get_next_unspect_bandwidth_voucher()
+            .get_next_unspent_ticketbook()
             .await)
     }
 
-    async fn consume_coconut_credential(
+    async fn consume_coconut_credential(&self, id: i64) -> Result<(), StorageError> {
+        self.coconut_credential_manager
+            .consume_coconut_credential(id)
+            .await;
+
+        Ok(())
+    }
+
+    async fn update_issued_credential<'a>(
```

Comment:
why do we need the whole thing

Reply:
Mentioned it earlier, internal counter needs updating